src/mito2/src/manifest/manager.rs

```diff
@@ -515,8 +522,18 @@ impl RegionManifestManager {
 
     /// Increases last version and returns the increased version.
     fn increase_version(&mut self) -> ManifestVersion {
-        self.last_version += 1;
-        self.last_version
+        let previous = self.last_version.fetch_add(1, Ordering::Relaxed);
+        previous + 1
```

Comment:
I think it's not appropriate to return the increased version like this, because the real value might be just changed after you called `fetch_add`. Instead, always get the value like
```suggestion
        self.last_version()
```
Same to the `set_version` below.